aiidalab_widgets_base/viewers.py

```diff
@@ -602,6 +603,7 @@ class StructureDataViewer(_StructureDataBaseViewer):
 
     structure = Union([Instance(Atoms), Instance(Node)], allow_none=True)
     displayed_structure = Instance(Atoms, allow_none=True, read_only=True)
+    pk = Int(allow_none=True)
```

Comment:
I don't think we need this trait (see comment above).

Reply:
Unfortunately, we do. See line 619 (validating structure). The structure always becomes an ASE object that does not contain the PK number.

Reply:
Hence why I'm checking whether the structure trait has the pk attribute in the other code segment.

Reply:
Got it.